components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +69,312 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
+                         constant:-kBottomPadding]
+          .active = YES;
+    }
+  }
+
+  [self updateButtonsConstraintsWithLayoutStyle:layoutStyle];
 }
 
-#pragma mark - Layout methods
+#pragma mark - Layout helpers
 
-- (void)applyMDCBannerViewLayout:(MDCBannerViewLayout *)layout {
-  if (![self isMDCBannerViewLayoutApplicable:layout]) {
-    return;
+- (void)updateButtonsConstraintsWithLayoutStyle:(MDCBannerViewLayoutStyle)layoutStyle {
+  if (@available(iOS 9.0, *)) {
+    if (self.trailingButton.hidden) {
+      [self.leadingButton.trailingAnchor
+          constraintEqualToAnchor:self.buttonContainerView.trailingAnchor]
+          .active = YES;
+      [self.leadingButton.centerYAnchor
+          constraintEqualToAnchor:self.buttonContainerView.centerYAnchor]
+          .active = YES;
+    } else {
+      if (layoutStyle == MDCBannerViewLayoutMultiLineStackedButtonStyle) {
+        [self.leadingButton.trailingAnchor
+            constraintEqualToAnchor:self.buttonContainerView.trailingAnchor]
+            .active = YES;
+        [self.trailingButton.topAnchor constraintEqualToAnchor:self.leadingButton.bottomAnchor
+                                                      constant:kButtonVerticalIntervalSpace]
+            .active = YES;
+      } else {
+        [self.leadingButton.trailingAnchor constraintEqualToAnchor:self.trailingButton.leadingAnchor
+                                                          constant:-kButtonHorizontalIntervalSpace]
+            .active = YES;
+      }
+      [self.leadingButton.topAnchor constraintEqualToAnchor:self.buttonContainerView.topAnchor]
+          .active = YES;
+      [self.leadingButton.leadingAnchor
+          constraintGreaterThanOrEqualToAnchor:self.buttonContainerView.leadingAnchor]
+          .active = YES;
+      [self.trailingButton.trailingAnchor
+          constraintEqualToAnchor:self.buttonContainerView.trailingAnchor]
+          .active = YES;
+      [self.trailingButton.bottomAnchor
+          constraintEqualToAnchor:self.buttonContainerView.bottomAnchor]
+          .active = YES;
+    }
   }
+}
 
-  self.iconImageViewContainer.frame = layout.imageContainerFrame;
-  self.iconImageView.center = CGPointMake(CGRectGetWidth(layout.imageContainerFrame) / 2,
-                                          CGRectGetHeight(layout.imageContainerFrame) / 2);
-  self.textLabel.frame = layout.textLabelFrame;
-  [self.buttons enumerateObjectsUsingBlock:^(id object, NSUInteger idx, BOOL *stop) {
-    if (idx >= layout.buttonFrames.count) {
-      *stop = YES;
-      return;
+- (MDCBannerViewLayoutStyle)styleForSizeToFit:(CGSize)sizeToFit {
+  MDCBannerViewLayoutStyle style;
+  CGFloat remainingWidth = sizeToFit.width;
+  CGFloat marginsPadding = self.layoutMargins.left + self.layoutMargins.right;
+  remainingWidth -= marginsPadding;
+  remainingWidth -= (kLeadingPadding + kTrailingPadding);
+  if (self.trailingButton.hidden) {
+    [self.leadingButton sizeToFit];
```

Comment:
you should move things that are in both branches of an if else outside of the conditional entirely

Reply:
Addressed in 070341d.